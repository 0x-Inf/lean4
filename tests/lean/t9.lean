precedence `+` : 65
precedence `*` : 75
precedence `=` : 50
precedence `≃` : 50
variable N : Type.{1}
variable a : N
variable b : N
variable add : N → N → N
variable mul : N → N → N
namespace foo
  infixl + : add
  infixl * : mul
  check a+b*a
end
-- Notation is not avaiable outside the namespace
check a+b*a
namespace foo
  -- Notation is restored
  check a+b*a
end
