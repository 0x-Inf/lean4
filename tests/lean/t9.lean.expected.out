add a (mul b a) : N
t9.lean:16:7: error: invalid expression
add a (mul b a) : N
